lang/src/org/partiql/lang/syntax/PartiQLParser.kt

```diff
@@ -46,7 +47,7 @@ import org.partiql.grammar.parser.generated.PartiQLTokens as GeneratedLexer
 internal class PartiQLParser(
     private val ion: IonSystem,
     val customTypes: List<CustomType> = listOf()
-) : Parser {
+) : Parser, org.partiql.lang.syntax.Lexer {
```

Comment:
The fact that Parser has to implement both Parser and Lexer seems a bit odd to me. Is it possible to decouple them by creating a `PartiQLLexer` class? I do not have a strong opinion on this.

Reply:
Yeah, I was going to leave it to the reviewer's discretion whether we should create a PartiQLLexer, and I like it. It's a small change -- will send it out tomorrow! Thanks for the feedback!